### ISSUE TITLE
Inability to hold multiple tapret tweaks in Tr RGB wallet for the same derivation

### ISSUE THREAD

**Issue by crisdut** [RGB-WG/rgb#183]
**Description**

I discovered a strange behavior when we tried to use `TapretKey::derive` after making a successful `tapret1st`. If I retry using the same terminal change of the previous transaction, the pay operation returns `TapretKeyError::TapTreeNonEmpty`:

```shell
use of taproot script descriptors is not yet supported. 
You may also check the latest version of the software which may already support this feature.
```

This occurs because when creating the PSBT, we use the `TapretKey::derive` inside of the `Psbt::construct_change` method (see [here](https://github.com/BP-WG/bp-std/blob/develop/psbt/src/data.rs#L424)), and derivation uses the tap tweak to build a taptree.

In version 0.10, we could derive without tap-tweak; I believe we need to implement something similar.

**Comment by zoedberg** [comment]
@crisdut could you please tell me on which branch you've found this issue?

**Comment by crisdut** [comment]
> @crisdut could you please tell me on which branch you've found this issue?

Hi @zoedberg ,

I first found it in beta5, but it remains in beta6 (master). It's easier to find it using rgb-rt as a lib and not in the cli.

**Comment by dr-orlovsky** [comment]
Yes, basically if you derive for a key which already has an associated at-tweak it will always derive it in a tweaked way. It is also the fact that currently we can only handle only single at-tweak per key... Both things should be changed, but it would require some deep refactoring of the whole descriptor and derivation APIs starting from `bp-std`, and I was planning it for v0.12...
